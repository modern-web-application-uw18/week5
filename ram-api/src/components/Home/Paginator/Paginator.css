.Paginator {

}

.pagination>li>a, .pagination>li>span {
    background-color: #000000;
    color: #dedede;
}