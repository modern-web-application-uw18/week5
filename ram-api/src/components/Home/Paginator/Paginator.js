import React from 'react';
import './Paginator.css';
import PropTypes from 'prop-types';


const paginator = (props) => {


        return (
            <nav aria-label="Page navigation">
                <ul className="pagination" style={{ opacity: 0.5 }}>

                    <li>
                        <a aria-label="Previous" onClick={(e) => props.paginatorHandler(e, "prev")}>
                            <span aria-hidden="true">&laquo;</span>
                        </a>
                    </li>

                    <li><a onClick={(e) => props.paginatorHandler(e, "inner", props.first)}>{props.first}</a></li>
                    <li><a onClick={(e) => props.paginatorHandler(e, "inner", props.second)}>{props.second}</a></li>
                    <li><a onClick={(e) => props.paginatorHandler(e, "inner", props.third)}>{props.third}</a></li>
                    <li><a onClick={(e) => props.paginatorHandler(e, "inner", props.fourth)}>{props.fourth}</a></li>
                    <li><a onClick={(e) => props.paginatorHandler(e, "inner", props.fifth)}>{props.fifth}</a></li>

                    <li>
                        <a aria-label="Next" onClick={(e) => props.paginatorHandler(e, "next")}>
                            <span aria-hidden="true">&raquo;</span>
                        </a>
                    </li>


                </ul>
            </nav>
        )
    
}


paginator.propTypes = {
    pageCount: PropTypes.number,
    characterCount: PropTypes.number,
    paginatorHandler: PropTypes.func,
    first: PropTypes.number,
    second: PropTypes.number,
    third: PropTypes.number,
    fourth: PropTypes.number,
    fifth: PropTypes.number
}


export default paginator;
