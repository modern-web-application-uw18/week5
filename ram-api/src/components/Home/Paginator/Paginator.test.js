import React from 'react';
import { mount } from 'enzyme';
import Paginator from './Paginator'
import { MemoryRouter } from 'react-router-dom';
// import character from '../../../__mocks__/character.json'

describe('<Paginator />', () => {
    it('Should render Paginator', () => {

        const wrapper = mount(
            // <MemoryRouter>
                <Paginator />
            // </MemoryRouter>
        );


        expect(wrapper).toMatchSnapshot();
    })
})




