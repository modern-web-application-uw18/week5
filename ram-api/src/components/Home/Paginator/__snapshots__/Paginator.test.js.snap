// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginator /> Should render Paginator 1`] = `
<paginator>
  <nav
    aria-label="Page navigation"
  >
    <ul
      className="pagination"
      style={
        Object {
          "opacity": 0.5,
        }
      }
    >
      <li>
        <a
          aria-label="Previous"
          onClick={[Function]}
        >
          <span
            aria-hidden="true"
          >
            «
          </span>
        </a>
      </li>
      <li>
        <a
          onClick={[Function]}
        />
      </li>
      <li>
        <a
          onClick={[Function]}
        />
      </li>
      <li>
        <a
          onClick={[Function]}
        />
      </li>
      <li>
        <a
          onClick={[Function]}
        />
      </li>
      <li>
        <a
          onClick={[Function]}
        />
      </li>
      <li>
        <a
          aria-label="Next"
          onClick={[Function]}
        >
          <span
            aria-hidden="true"
          >
            »
          </span>
        </a>
      </li>
    </ul>
  </nav>
</paginator>
`;
